src/sshfs_mount/sshfs_mount.cpp

```diff
@@ -28,38 +28,95 @@
     if (ssh_process.exit_code() != 0)
-        error_handler(ssh_process);
-    return ssh_process.read_std_output();
-}
+        throw std::runtime_error(ssh_process.read_std_error());
 
-// Run a command on a given SSH session.
-auto run_cmd(mp::SSHSession& session, std::string&& cmd)
-{
-    auto error_handler = [](mp::SSHProcess& proc) { throw std::runtime_error(proc.read_std_error()); };
-    return run_cmd(session, std::forward<std::string>(cmd), error_handler);
+    return ssh_process.read_std_output() + ssh_process.read_std_error();
 }
 
-// Check if sshfs exists on a given SSH session.
-void check_sshfs_exists(mp::SSHSession& session)
+auto get_sshfs_exec_and_options(mp::SSHSession& session)
 {
-    auto error_handler = [](mp::SSHProcess& proc) {
-        mpl::log(mpl::Level::warning, category,
-                 fmt::format("Unable to determine if 'sshfs' is installed: {}", proc.read_std_error()));
-        throw mp::SSHFSMissingError();
-    };
+    std::string sshfs_exec;
+
+    try
+    {
+        // Prefer to use Snap package version first
+        auto sshfs_env{run_cmd(session, "snap run multipass-sshfs.env")};
+
+        auto ld_library_path = mp::utils::match_line_for(sshfs_env, ld_library_path_key);
+        auto snap_path = mp::utils::match_line_for(sshfs_env, snap_path_key);
+        snap_path = snap_path.substr(snap_path_key.length());
+
+        sshfs_exec = fmt::format("env {} {}/bin/sshfs", ld_library_path, snap_path);
+    }
+    catch (const std::exception& e)
+    {
+        mpl::log(mpl::Level::debug, category,
+                 fmt::format("'multipass-sshfs' snap package is not installed: {}", e.what()));
+
+        // Fallback to looking for distro version if snap is not found
+        try
+        {
+            sshfs_exec = run_cmd(session, "sudo which sshfs");
+        }
+        catch (const std::exception& e)
+        {
+            mpl::log(mpl::Level::warning, category,
+                     fmt::format("Unable to determine if 'sshfs' is installed: {}", e.what()));
+            throw mp::SSHFSMissingError();
+        }
+    }
+
+    sshfs_exec = mp::utils::trim_end(sshfs_exec);
+
+    auto version_info{run_cmd(session, fmt::format("sudo {} -V", sshfs_exec))};
+
+    sshfs_exec += " -o slave -o transform_symlinks -o allow_other";
+
+    auto fuse_version_line = mp::utils::match_line_for(version_info, fuse_version_string);
+    if (!fuse_version_line.empty())
+    {
+        std::string fuse_version;
+
+        // split on the fuse_version_string along with 0 or 1 colon(s)
+        auto tokens = mp::utils::split(fuse_version_line, fmt::format("{}:? ", fuse_version_string));
+        if (tokens.size() == 2)
+            fuse_version = tokens[1];
+
+        if (fuse_version.empty())
+        {
+            mpl::log(mpl::Level::warning, category, fmt::format("Unable to parse the {}", fuse_version_string));
+            mpl::log(mpl::Level::debug, category, fmt::format("Value is {}", fuse_version_line));
```

Comment:
```suggestion
            mpl::log(mpl::Level::debug, category, fmt::format("Unable to parse the {}: {}", fuse_version_string, fuse_version_line));
```

In case you filter for just debug messages, or something else logs in between (I know, unlikely)?

Reply:
Ok, sure.